Tests/Tests/STPPaymentMethodTupleTest.m

```diff
@@ -0,0 +1,96 @@
+}
+
+- (void)testNotAllowedAvailableSelectedPaymentMethod {
+
+    STPCard *card = [STPCard decodedObjectFromAPIResponse:[STPTestUtils jsonNamed:@"Card"]];
+    STPSource *source = [STPSource decodedObjectFromAPIResponse:[STPTestUtils jsonNamed:@"CardSource"]];
+
+    STPPaymentMethodTuple *tuple = [[STPPaymentMethodTuple alloc] initWithSavedPaymentMethods:@[card, source]
+                                                                        availablePaymentTypes:@[[STPPaymentMethodType creditCard],
+                                                                                                [STPPaymentMethodType giropay]]
+                                                                        selectedPaymentMethod:[STPPaymentMethodType creditCard]];
+
+    XCTAssertNil(tuple.selectedPaymentMethod);
+}
+
+- (void)testMissingSelectedPaymentMethod {
+
+    STPCard *card = [STPCard decodedObjectFromAPIResponse:[STPTestUtils jsonNamed:@"Card"]];
+    STPSource *source = [STPSource decodedObjectFromAPIResponse:[STPTestUtils jsonNamed:@"CardSource"]];
+
+    STPPaymentMethodTuple *tuple = [[STPPaymentMethodTuple alloc] initWithSavedPaymentMethods:@[card, source]
+                                                                        availablePaymentTypes:@[[STPPaymentMethodType creditCard],
+                                                                                                [STPPaymentMethodType giropay]]
+                                                                        selectedPaymentMethod:[STPPaymentMethodType sofort]];
+
+    XCTAssertNil(tuple.selectedPaymentMethod);
+}
+
+- (void)testOnlyOneAvailableUnselectablePaymentMethod {
+    STPPaymentMethodTuple *tuple = [[STPPaymentMethodTuple alloc] initWithSavedPaymentMethods:nil
+                                                                        availablePaymentTypes:@[[STPPaymentMethodType creditCard]]
+                                                                        selectedPaymentMethod:nil];
+
+    XCTAssertNil(tuple.selectedPaymentMethod);
+}
+
+- (void)testOnlyOneAvailableNotAllowedPaymentMethod {
+    STPPaymentMethodTuple *tuple = [[STPPaymentMethodTuple alloc] initWithSavedPaymentMethods:nil
+                                                                        availablePaymentTypes:@[[STPPaymentMethodType creditCard]]
+                                                                        selectedPaymentMethod:nil];
+
+    XCTAssertNil(tuple.selectedPaymentMethod);
+}
+
+- (void)testOnlyOneAvailableAllowedPaymentMethod {
+    STPPaymentMethodTuple *tuple = [[STPPaymentMethodTuple alloc] initWithSavedPaymentMethods:nil
+                                                                        availablePaymentTypes:@[[STPPaymentMethodType bancontact]]
+                                                                        selectedPaymentMethod:nil];
+
+    XCTAssertEqualObjects(tuple.selectedPaymentMethod, [STPPaymentMethodType bancontact]);
```

Comment:
ah interesting. I guess this behavior makes sense, if the customer doesn't have a `defaultSource` we shouldn't auto-select a saved card.